### DISCUSSION TITLE
Basics: How to map states to values on plot? (Advise needed).

### DISCUSSION THREAD

**Discussion by mysleinic** [dbuezas/lovelace-plotly-graph-card#356] [Show and tell]
Hi All,

I'm struggling to find some example on how to map states to values on the plot?

![image](https://github.com/dbuezas/lovelace-plotly-graph-card/assets/139065142/45277e89-c834-46b6-a4fe-4a6cb8c7d8f4)

I would like to map states "on", "off" and "heat" on that plot to numbers - "0" or "1".

In order to have always the same sequence... sometimes when scrolling on is on the bottom, sometimes on top.. it's confusing.

ps. Excuse me for such basic questions, but all of the examples here are complex topics for a magnificent great looking stuff... 
I would like to keep it simple as that... I didn't find exact examples, all my experiments failed...

**Comment by dbuezas** [comment]
Hi there!
You can also set a fix order of the categorical axis (non numbers)

```yaml
layout:
    xaxis:
        categoryorder: array
        categoryarray:  ["on", ”off", "heat"]
```

To map values see the `map_y` filter in the readme.

Something like `map_y: "y == 'on' ? 0 : y == 'off' ? 1 : y == 'heating' ? 2 : undefined"`

**Reply by dbuezas** [reply]
Ps: within the plotting cards available, this is on the "advanced" side of things. This means you can do pretty much anything, but it has a steeper learning curve if you don't have a programming background.

That being said, I hope you enjoy the learning process! :)

**Reply by mysleinic** [reply]
Thank you so much - I was so close on my own with `map_y` filter... little syntax mistake + I didn't know how to map multiple states.
Both of what you have suggested is working, however I decided to use axis category options. Applied to my y3 axis did the charm!
All in nice order looking just perfectly fine to allow me troubleshoot and tune PID regulators in my apartment heating system ;) 

Thank you so much for taking your precious time, to help some random noob (me) with such a basic question!


ps. I used to program a bit of a different stuff - Turbo Pascal / Delphi, PLC controllers, ATMega microcontrolers in assembler.. a power plant DCS systems, some config yamls in web application... **BUT** overall... I would say I have some to little programming experience. More over **I'm doing my best to stay at this level!** ;)

**Reply by dbuezas** [reply]
Cool! Happy to hear you succeeded!

Regarding the mapping, google " javascript ternary operator". You can also use a switch case statement if you have to map more states.

Oh, you do have coding experience, just google "how to do xyz in JavaScript" and you'll get there quickly!

**Reply by mysleinic** [reply]
Thanks - I will check the next time :)
